### PR TITLE
2.0 prep: Remove deprecated flags

### PR DESCRIPTION
**Pull request by jsafrane**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In 2.0 we can safely remove deprecated flags.

**Which issue(s) this PR fixes**:
Related to  #158

**Does this PR introduce a user-facing change?**:

```release-note
Action required: the `-connection-timeout`, `-leader-election-type` and `-leader-election-identity` flags, deprecated in v1.2, have been removed. Please update your manifests for the external-attacher.
Action required: the `-dummy` flag has been removed. Please update your manifests for the external-attacher.
```

### CODE REVIEW

File: cmd/csi-attacher/main.go [hunk 1]
```diff
@@ -57,21 +56,18 @@ const (
...
-	resync            = flag.Duration("resync", 10*time.Minute, "Resync interval of the controller.")
-	connectionTimeout = flag.Duration("connection-timeout", 0, "This option is deprecated.")
-	csiAddress        = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
-	dummy             = flag.Bool("dummy", false, "Run in dummy mode, i.e. not connecting to CSI driver and marking everything as attached. Expected CSI driver name is \"csi/dummy\".")
-	showVersion       = flag.Bool("version", false, "Show version.")
-	timeout           = flag.Duration("timeout", 15*time.Second, "Timeout for waiting for attaching or detaching the volume.")
+	kubeconfig  = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	resync      = flag.Duration("resync", 10*time.Minute, "Resync interval of the controller.")
+	csiAddress  = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
+	dummy       = flag.Bool("dummy", false, "Run in dummy mode, i.e. not connecting to CSI driver and marking everything as attached. Expected CSI driver name is \"csi/dummy\".")
```
Comment (msau42) [inline cmd/csi-attacher/main.go]: Was this only for initial development? Can we remove this too?

Comment (jsafrane) [reply cmd/csi-attacher/main.go]: I removed `-dummy` too, I hope release note generator can handle multi-line comments :-)

**Comment by jsafrane** [comment]
I removed `-dummy`. And when at it, I noticed that there was an old unused variable with the old CRD-based informers, so I deleted it (removing lot of files from vendor/)

**Pull request merged**